src/components/networked.js

```diff
@@ -17,6 +17,60 @@ function defaultRequiresUpdate() {
   };
 }
 
+AFRAME.registerSystem("networked", {
+  init() {
+    this.updateNextSyncTime();
+    this.components = [];
+  },
+
+  register(component) {
+    this.components.push(component);
+  },
+
+  deregister(component) {
+    const idx = this.components.indexOf(component);
+
+    if (idx > -1) {
+      this.components.splice(idx, 1);
+    }
+  },
+
+  tick: (function() {
+    const data = { d: [] };
+
+    return function() {
+      if (!NAF.connection.adapter) return;
+      if (!this.needsToSync()) return;
+
+      for (let i = 0, l = this.components.length; i < l; i++) {
+        const c = this.components[i];
+        if (!c.isMine()) continue;
+        if (!c.el.parentElement) continue;
+
+        const syncData = this.components[i].syncDirty();
+        if (!syncData) continue;
+
+        data.d.unshift(syncData);
```

Comment:
Why unshift instead of push? I think this will flip the order things are updated since components tick in the order they were created and would have broadcasted in that same order previously. They are registered in creation order but would be iterated in reverse order on the receiving end now.

Reply:
oof, i had `push` earlier but i thought that was backwards. i was seeing weird behavior on my other branch and thought that was the bug. nice catch.